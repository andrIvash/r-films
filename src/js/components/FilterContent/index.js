// @flow
import React from 'react';

type Props = {
 onFilterSelect: (ev: SynteticInputIvent) => void,
 selected: string
}

const FilterContent = (props: Props) => (
  <div className='filter-content'>
    <div className='filter-content__title'>Sort By</div>
    <div className='filter-content__group'>
      <label
        className={props.selected === 'release' ?
        'filter-content__label active' :
        'filter-content__label'}
        htmlFor='release'
        >
        <input
          autoComplete='off'
          className='filter-content__input'
          id='release'
          name='filter'
          onChange={(ev) => props.onFilterSelect(ev.target.value)}
          type='radio'
          value='release'
        /> release date
      </label>
      <label
        className={props.selected === 'rating' ?
        'filter-content__label active' :
        'filter-content__label'}
        htmlFor='rating'
        >
        <input
          autoComplete='off'
          className='filter-content__input'
          id='rating'
          name='filter'
          onChange={(ev) => props.onFilterSelect(ev.target.value)}
          type='radio'
          value='rating'
        /> rating
      </label>
    </div>
  </div>
);

export default FilterContent;
