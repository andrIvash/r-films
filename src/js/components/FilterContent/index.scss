.filter-content {
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-family: $sign-font;
  font-size: rem(16px);
  font-weight: 500;
  color: $color-background;
}

.filter-content__label {
  margin-bottom: 0;
  margin-left: rem(15px);
  cursor: pointer;
}

.filter-content__label.active {
  color: $color-logo;
}

.filter-content__input {
  display: none;
}