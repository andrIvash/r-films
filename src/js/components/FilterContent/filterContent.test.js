import React from 'react';
import FilterContent from './index';

const props = {};

describe('FilterContent', () => {
  it('should be defined', () => {
    expect(FilterContent).toBeDefined();
  });
  it('should render correctly', () => {
    const wrapper = shallow(
      <FilterContent {...props} /> );
    expect(wrapper).toMatchSnapshot();
  });

  it(`should calls onFilterSelect 
    when input is selected`, () => {
    const props = {
      onFilterSelect: jest.fn(),
      selected: 'release',
    };
    const wrapper = shallow(
      <FilterContent {...props} /> );
    wrapper.find('#release').simulate('change');
    expect(props.onFilterSelect).toHaveBeenCalledTimes(1);
    expect(wrapper.find('#release').parent().hasClass('active')).toBe(true);
  });

  it(`should calls onFilterSelect 
    when rating input is selected`, () => {
    const props = {
      onFilterSelect: jest.fn(),
      selected: 'rating',
    };
    const wrapper = shallow(
      <FilterContent {...props} /> );
    wrapper.find('#rating').simulate('change');
    expect(props.onFilterSelect).toHaveBeenCalledTimes(1);
    expect(wrapper.find('#rating').parent().hasClass('active')).toBe(true);
  });
});
