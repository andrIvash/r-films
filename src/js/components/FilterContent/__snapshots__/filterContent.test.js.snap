// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterContent should render correctly 1`] = `
<div
  class="filter-content"
>
  <div
    class="filter-content__title"
  >
    Sort By
  </div>
  <div
    class="filter-content__group"
  >
    <label
      class="filter-content__label active"
      for="release"
    >
      <input
        autocomplete="off"
        class="filter-content__input"
        id="release"
        name="filter"
        type="radio"
        value="release"
      />
       release date
    </label>
    <label
      class="filter-content__label"
      for="rating"
    >
      <input
        autocomplete="off"
        class="filter-content__input"
        id="rating"
        name="filter"
        type="radio"
        value="rating"
      />
       rating
    </label>
  </div>
</div>
`;
